src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (true) {
+      case action.type === 'addProperties' :
+        Object.assign(state, action.extraData);
+        break;
+
+      case action.type === 'removeProperties' :
+        for (const keys of action.keysToRemove) {
+          if (state.hasOwnProperty(keys)) {
```

Comment:
It is overkill, if you delete property, which wasn't in object nothing will change